scripts/devops_tasks/python_snippet_updater.py

```diff
@@ -0,0 +1,94 @@
+import sys
+from pathlib import Path
+import re
+
+snippets = {}
+not_up_to_date = False
+
+target_snippet_sources = ["samples/*.py", "samples/*/*.py"]
+target_md_files = ["README.md"]
+
+def get_snippet(file):
+    with open(file, 'r') as f:
+        content = f.read()
+    pattern = "# \\[START[A-Z a-z0-9_]+\\][\\s\\S]+?# \\[END[A-Z a-z0-9_]+\\]"
+    matches = re.findall(pattern, content)
+    for match in matches:
+        s = match
+        pos1 = s.index("[START")
+        pos2 = s.index("]")
+        name = s[pos1 + 6:pos2].strip()
+        s = s[pos2 + 1:]
+        pos1 = s.index("# [END")
+        snippet = s[:pos1-1]
+        # Remove extra spaces
+        spaces = ""
+        for char in snippet[1:]:
+            if char == " ":
+                spaces += char
+            else:
+                break
+        snippet = snippet.replace("\n" + spaces, "\n")
+        # Remove first newline
+        snippet = snippet[1:].rstrip()
+        if snippet[-1] == "\n":
+            snippet = snippet[:-1]
+
+        file_name = str(file.name)[:-3]
+        identifier = ".".join([file_name, name])
+        if identifier in snippets.keys():
+            print(f'Warning: found duplicated snippet name "{identifier}".')
+            print(file)
+        # print(f"Found: {file.name}.{name}")
+        snippets[identifier] = snippet
+
+
+def update_snippet(file):
+    with open(file, 'r') as f:
+        content = f.read()
+    pattern = "<!-- SNIPPET:[A-Z a-z0-9_.]+-->[\\s\\S]*?<!-- END SNIPPET -->"
+    matches = re.findall(pattern, content)
+    for match in matches:
+        s = match
+        pos1 = s.index("-->")
+        header = s[:pos1+3]
+        name = s[13:pos1].strip()
+        # print(f"Info: found name: {name}")
+        if name not in snippets.keys():
+            print(f'Warning: cannot found snippet name "{name}".')
+            exit(1)
+        target = "".join([header, "\n```python\n", snippets[name], "\n```\n", "<!-- END SNIPPET -->"])
+        if s != target:
+            print(f'Warning: snippet "{name}" is not up to date.')
+            global not_up_to_date
+            not_up_to_date = True
+            content = content.replace(s, target)
+    with open(file, 'w') as f:
+        f.write(content)
+
+
+if __name__ == "__main__":
+    arg_len = len(sys.argv)
```

Comment:
Use `argparse`. I won't block on this but definitely use `argparse`.

[Example in our own](https://github.com/Azure/azure-sdk-for-python/blob/b0982ba999c3fd1188ba44c3ca5e2e4963d34863/scripts/devops_tasks/use_pypy_version.py#L66)